### ISSUE TITLE
Package maap-py so it can be installed through a package manager

### ISSUE THREAD

**Issue by marjo-luc** [MAAP-Project/Community#801]
Right now, maap-py needs to be installed from source. This makes it harder on users who are creating custom environments who may not know how or which versions to install from source. Having the module packaged and installable using a package manager such as pip would improve the user experience and align it with how users install other packages.

**Comment by grallewellyn** [comment]
@marjo-luc Do you want me to work on this? I am looking for new tickets. Is pip the preferred way to go?

**Comment by rtapella** [comment]
we did this for the Unity project in case you want to talk to ppl about how to get it into pypi
https://pypi.org/project/unity-sds-client/

**Comment by rtapella** [comment]
One thing you should do first is look at pypi to see if there are similarly named packages...
https://pypi.org/project/MAAP/

**Comment by rtapella** [comment]
(some of the MAAP stuff is from Marjorie, it looks like... but not all?)

**Comment by grallewellyn** [comment]
Thanks Rob! I was working on this for ipycmc following these steps: https://packaging.python.org/en/latest/tutorials/packaging-projects/ 

Looks like I can get away with the name maap-py

**Comment by marjo-luc** [comment]
We have the jupyter server extension on pypi already. The frontend extensions are on npm.

Let's hold off on doing this right away. I want to take this opportunity to review maap-py first.

**Comment by grallewellyn** [comment]
I believe this includes removing the need for locating the maap.cfg file to instantiate a MAAP instance, correct me if I am wrong since that is not a separate ticket but I can make one

**Comment by marjo-luc** [comment]
We would like to do that, but that doesn't necessarily preclude us from packaging maap-py. Feel free to create a ticket to track maap.cfg updates separately.

**Comment by grallewellyn** [comment]
Created new ticket here: https://github.com/MAAP-Project/Community/issues/909

**Comment by chuckwondo** [comment]
@grallewellyn, we can package maap.cfg within the maap-py package as part of the packaging process. To aid doing so, it should be moved within the maap directory so that it becomes a package resource. It's a matter of only a couple of code changes related to reading the config and an addition to setup.py.

I'd be happy to show you how to do this.

**Comment by grallewellyn** [comment]
@sujen1412 said he wanted to work on the ticket for removing the need for tracking maap.cfg

**Comment by chuckwondo** [comment]
@grallewellyn, as mentioned in conversation with @sujen1412 in https://github.com/MAAP-Project/Community/issues/909, I'll now work on making `maap.cfg` a package resource, which will obviate the need for using the `MAAP_CONF` env var. This corresponding issue in the maap-py repo: https://github.com/MAAP-Project/maap-py/issues/78.

**Comment by chuckwondo** [comment]
I believe my work on https://github.com/MAAP-Project/maap-py/issues/78 addressed this issue. Does that suffice to close this issue as complete?

**Comment by grallewellyn** [comment]
@chuckwondo Did you release maap-py on a package manager? I don't see it on pip, conda or npm
I think you did resolve this issue though: https://github.com/MAAP-Project/Community/issues/909

**Comment by chuckwondo** [comment]
Nope, no release. Isn't @bsatoriu supposed to work on setting things up to allow us to cut releases to pip and/or conda-forge?

**Comment by marjo-luc** [comment]
[https://github.com/MAAP-Project/maap-py/issues/78](https://github.com/MAAP-Project/maap-py/issues/78) does not satisfy the work captured in this ticket.

@chuckwondo I'm not aware of any other ticket that describes the work outlined in this ticket. If there is one, please feel free to link it to this one.

**Comment by chuckwondo** [comment]
@marjo-luc and @grallewellyn, apologies, I replied to the wrong issue. I meant to comment in #909, which I'll do now.

**Comment by chuckwondo** [comment]
As we discussed in a [Slack thread](https://teamdsig.slack.com/archives/C01G0NFAHDJ/p1709302528079589), I'm adding a dependency on [removing support for Python 2](https://github.com/MAAP-Project/maap-py/issues/34). We'll wait to publish any sort of package until Python 2 remnants are removed, and a minimum Python 3.x version is documented.

**Comment by chuckwondo** [comment]
@marjo-luc, I just chatted with @grallewellyn about prioritizing this issue of packaging/publishing maap-py over other things and she mentioned that you seem to be on the same page. She also indicated that you'v recently been packaging and publishing other things as well, so do you want to do it for maap-py, or do you need someone else to pick up this issue?

**Comment by marjo-luc** [comment]
@chuckwondo the platform team will discuss this during our release planning tomorrow.
